### ISSUE TITLE
Keras Model Import Does Not Support Class Handles for Loss Functions

### ISSUE THREAD

**Issue by alexdriedger** [deeplearning4j/deeplearning4j#8990]
#### Issue Description

`MeanAbsoluteError` is not a supported Keras loss function. I get the same error for `MeanSquaredError` and `MeanSquaredLogarithmicError` 

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version = 1.0.0-beta7
* Platform information (OS, etc) = Windows 10
* CUDA version, if used = N/A
* NVIDIA driver version, if in use = N/A

#### Additional Information

Where applicable, please also provide:

```
org.deeplearning4j.nn.modelimport.keras.exceptions.UnsupportedKerasConfigurationException: Unknown Keras loss function MeanAbsoluteError. Please file an issue at https://github.com/eclipse/deeplearning4j/issues.
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasLossUtils.mapLossFunction(KerasLossUtils.java:100)
	at org.deeplearning4j.nn.modelimport.keras.layers.KerasLoss.<init>(KerasLoss.java:86)
	at org.deeplearning4j.nn.modelimport.keras.layers.KerasLoss.<init>(KerasLoss.java:64)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.importTrainingConfiguration(KerasModel.java:269)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:140)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:57)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildSequential(KerasModelBuilder.java:322)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:223)
	at Main.main(Main.java:23)

```

Here is the original python code which generated the model

```python
model = tf.keras.models.Sequential([
  tf.keras.layers.Dense(256, input_shape=(975,), activation='relu'),
  tf.keras.layers.Dropout(.2),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(.2),
  tf.keras.layers.Dense(64, activation='relu'),
  tf.keras.layers.Dense(16, activation='relu'),
  tf.keras.layers.Dense(1)
])

model.summary()
model.compile(optimizer=keras.optimizers.RMSprop(learning_rate=.001), loss=tf.keras.losses.MeanAbsoluteError(), metrics=["accuracy"])
```

**Comment by alexdriedger** [comment]
`MeanSquaredError` is also not supported. Is there an easy way to see which keras loss functions are supported besides recompiling my model, exporting, and loading it back into dl4j?

**Comment by treo** [comment]
That's weird. In principle we support pretty much all of the out of the box loss functions:
https://github.com/eclipse/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/config/KerasLayerConfiguration.java#L341-L360

What version of tf are you using here?

**Comment by alexdriedger** [comment]
Tensorflow 2.2.0

**Comment by alexdriedger** [comment]
I tried running it in the [standalone sample project](https://github.com/eclipse/deeplearning4j-examples/tree/master/standalone-sample-project) and I got the same error. I'm running Java 8.

This is the code I'm running that causes the error:

```java
try {
    String simpleMlp = new ClassPathResource("STSFP.h5").getFile().getPath(); // STSFP.h5 is generated in the above python code
    MultiLayerNetwork mymodel = KerasModelImport.importKerasSequentialModelAndWeights(simpleMlp);
    System.out.println(mymodel.summary());
} catch (Exception e) {
    e.printStackTrace();
}
```

**Comment by alexdriedger** [comment]
@treo It looks like the problem is not handling the case of using the keras loss function object instead of the String name of the loss function

## On the Java side

The exception is thrown in `mapLossFunction`

https://github.com/eclipse/deeplearning4j/blob/37d880a23cf7061d560fe8f77002b840c4019ca6/deeplearning4j/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLossUtils.java#L69-L71


Because `conf.getKERAS_LOSS_MEAN_ABSOLUTE_ERROR()` returns `"mean_absolute_error"` but `kerasLoss` is `"MeanAbsoluteError"`.

https://github.com/eclipse/deeplearning4j/blob/b5f0ec072f3fd0da566e32f82c0e43ca36553f39/deeplearning4j/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/config/KerasLayerConfiguration.java#L339

## What caused the error (on the Python side)

I was using `loss=tf.keras.losses.MeanAbsoluteError()` in `model.compile()` and it seems like dl4j currently only supports naming the loss function with a string, like `loss='mean_absolute_error'`.

Changing

```python
model.compile(optimizer=keras.optimizers.RMSprop(learning_rate=.001), loss=tf.keras.losses.MeanAbsoluteError(), metrics=["accuracy"])
```

to

```python
model.compile(optimizer=keras.optimizers.RMSprop(learning_rate=.001), loss='mean_absolute_error', metrics=["accuracy"])
```

Made the problem go away

**Comment by Adonis511** [comment]
However, I use myModel.compile(optimizer=opt,loss="mean_squared_error"), it doesn't word

**Comment by agibsonccc** [comment]
Closing this. This appears to have been fixed long ago. @Adonis511 please open a new issue with a reproducer. Your "it doesn't work" means nothing to me and will just be ignored. I can't read your computer or your mind. I need code with a reproducer and an error I can see if you want help. Otherwise it's just a waste of my time.
